src/main/java/net/glowstone/command/minecraft/TestForCommand.java

```diff
@@ -0,0 +1,88 @@
+package net.glowstone.command.minecraft;
+
+import net.glowstone.command.CommandTarget;
+import net.glowstone.command.CommandUtils;
+import net.glowstone.entity.GlowEntity;
+import net.glowstone.entity.GlowPlayer;
+import net.glowstone.io.entity.EntityStorage;
+import net.glowstone.util.mojangson.Mojangson;
+import net.glowstone.util.mojangson.ex.MojangsonParseException;
+import net.glowstone.util.nbt.CompoundTag;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.command.CommandSender;
+import org.bukkit.command.defaults.VanillaCommand;
+import org.bukkit.entity.Entity;
+
+import java.util.*;
+
+public class TestForCommand extends VanillaCommand {
+    public TestForCommand() {
+        super("testfor",
+                "Tests for a certain target in game",
+                "/testfor <player> [dataTag]",
+                Collections.emptyList());
+        setPermission("minecraft.command.testfor");
+    }
+
+    @Override
+    public boolean execute(CommandSender sender, String label, String[] args) {
+        if (!testPermission(sender)) {
+            return false;
+        }
+
+        if (args.length < 1) {
+            sender.sendMessage(ChatColor.RED + "Usage:" + usageMessage);
+            return false;
+        }
+
+        String name = args[0];
+        Entity[] entities;
+        if (name.startsWith("@")) {
+            CommandTarget target = new CommandTarget(sender, name);
+            entities = target.getMatched(CommandUtils.getLocation(sender));
+
+            if (entities.length == 0) {
+                sender.sendMessage(ChatColor.RED + "Selector '" + name + "' found nothing.");
+                return false;
+            }
+        } else {
+            GlowPlayer player = (GlowPlayer) Bukkit.getPlayerExact(args[0]);
+            if (player == null) {
+                sender.sendMessage(ChatColor.RED + "Entity '" + name + "' cannot be found");
+                return false;
+            } else {
+                entities = new Entity[] {player};
+            }
+        }
+
+        if (args.length >= 2) {
+            String data = String.join(" ", Arrays.copyOfRange(args, 4, args.length));
```

Comment:
The `4` here is incorrect and produces this exception if the number of arguments is under 4:

```
org.bukkit.command.CommandException: Unhandled exception executing 'testfor @e[type=pig] {Saddle:1b}' in net.glowstone.command.minecraft.TestForCommand(testfor)
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:160)
Caused by: java.lang.IllegalArgumentException: 4 > 2
        at java.util.Arrays.copyOfRange(Arrays.java:3480)
        at java.util.Arrays.copyOfRange(Arrays.java:3441)
        at net.glowstone.command.minecraft.TestForCommand.execute(TestForCommand.java:60)
```